src/lib/RequestParser.js

```diff
@@ -0,0 +1,202 @@
+     * @returns {string[]}
+     */
+    parsePathsArray(paths, name) {
+        if (!paths || paths.constructor !== Array) {
+            throw new Errors.InvalidRequestError(`${name} is required`);
+        }
+        if (paths.length === 0) {
+            throw new Errors.InvalidRequestError(`${name} must not be empty`);
+        }
+        /** @type {string[]} */
+        const requestedKeyPaths = [];
+        paths.forEach((/** @type {any} */path) => { // eslint-disable-line arrow-parens
+            if (typeof path !== 'string') {
+                throw new Errors.InvalidRequestError(`${name}: path must be of type string`);
+            }
+            requestedKeyPaths.push(this.parsePath(path, name));
+        });
+        return requestedKeyPaths;
+    }
+
+    /**
+     * @param {any} label
+     * @returns {string | undefined}
+     */
+    parseLabel(label) {
+        if (!label) {
+            return undefined;
+        }
+        if (typeof label !== 'string') {
+            throw new Errors.InvalidRequestError('Label must be of type string');
+        }
+        if (label.length === 0) {
+            return undefined;
+        }
+        return label;
+    }
+
+    /**
+     * @param {any} keyId
+     * @returns {Promise<KeyInfo>}
+     */
+    async parseKeyId(keyId) {
+        if (!keyId) {
+            throw new Errors.InvalidRequestError('keyId is required');
+        }
+        if (typeof keyId !== 'string') {
+            throw new Errors.InvalidRequestError('keyId must be of type string');
+        }
+        const keyInfo = await KeyStore.instance.getInfo(keyId);
+        if (!keyInfo) {
+            throw new Errors.KeyNotFoundError();
+        }
+        return keyInfo;
+    }
+
+    /**
+     * @param {any} indicesArray
+     * @returns {string[]}
+     */
+    parseIndicesArray(indicesArray) {
+        if (!indicesArray || indicesArray.constructor !== Array) {
+            throw new Errors.InvalidRequestError('indicesToDerive is required');
+        }
+        if (indicesArray.length === 0) {
+            throw new Errors.InvalidRequestError('Indice array must not be empty');
+        }
+        indicesArray.forEach((/** @type {any} */index) => { // eslint-disable-line arrow-parens
+            if (typeof index !== 'string') {
+                throw new Errors.InvalidRequestError('Each index of indicesToDerive must be a string.');
+            }
+            if (!index.endsWith("'")) {
+                throw new Errors.InvalidRequestError('Each index of IndicesToDerive must end with a \'.');
+            }
+            if (`${(parseInt(index.substr(0, index.length - 1), 10))}'` !== index) {
+                throw new Errors.InvalidRequestError('Each index of indicesToDerive must start with a number.');
+            }
+        });
+        return indicesArray;
+    }
+
+    /**
+     * @param {any} request
+     * @returns {Nimiq.ExtendedTransaction}
+     */
+    parseTransaction(request) {
```

Comment:
request => transaction

Reply:
I renamed it to object for now. Currently it is given the whole request in order to not have to build an additional object. In the future it might change, but transaction I feel is to restrictive.